sphinx_gallery/tests/test_full.py

```diff
@@ -241,18 +241,42 @@ def test_junit(sphinx_app, tmp_path):
     with open(junit_file, "rb") as fid:
         suite = lxml.etree.fromstring(fid.read())
     # this time we only ran the stale files
-    want.update(failures="2", skipped="1", tests="3")
+    want.update(failures="2", skipped="3", tests="5")
     got = dict(suite.attrib)
     del got["time"]
-    assert len(suite) == 3
-    assert suite[0].attrib["classname"] == "plot_numpy_matplotlib"
-    assert suite[0][0].tag == "failure", suite[0].attrib["classname"]
-    assert suite[0][0].attrib["message"].startswith("RuntimeError: Forcing")
-    assert suite[1].attrib["classname"] == "plot_scraper_broken"
-    assert suite[1][0].tag == "skipped", suite[1].attrib["classname"]
-    assert suite[2].attrib["classname"] == "plot_future_imports_broken"
-    assert suite[2][0].tag == "failure", suite[2].attrib["classname"]
-    assert suite[2][0].attrib["message"] == "Passed even though it was marked to fail"
+    skips_and_fails = [
+        {
+            "classname": "plot_failing_example",
+            "tag": "skipped",
+            "message": None,
+        },
+        {
+            "classname": "plot_failing_example_thumbnail",
+            "tag": "skipped",
+            "message": None,
+        },
+        {
+            "classname": "plot_numpy_matplotlib",
+            "tag": "failure",
+            "message": "RuntimeError: Forcing",
+        },
+        {
+            "classname": "plot_scraper_broken",
+            "tag": "skipped",
+            "message": None,
+        },
+        {
+            "classname": "plot_future_imports_broken",
+            "tag": "failure",
+            "message": "Passed even though it was marked to fail",
+        },
+    ]
+    assert len(suite) == len(skips_and_fails)
+    for this_suite, this_example in zip(suite, skips_and_fails):
+        assert this_suite.attrib["classname"] == this_example["classname"]
+        assert this_suite[0].tag == this_example["tag"], this_suite.attrib["classname"]
+        if this_example["message"] is not None:
+            assert this_suite[0].attrib["message"].startswith(this_example["message"])
```

Comment:
Needed to add new items for the 2 new examples. Thought perhaps this was easier to read.